### ISSUE TITLE
Can't build with Maven

### ISSUE THREAD

**Issue by brunoborges** [mspnp/spark-monitoring#127]
I'm trying to build the project with the following command:

`mvn install -P "scala-2.12_spark-3.0.1"`

And I get the following error:

```
[INFO] Reactor Summary for spark-monitoring 1.0.0:
[INFO]
[INFO] spark-monitoring ................................... FAILURE [  0.566 s]
[INFO] spark-listeners_3.0.1_2.12 ......................... SKIPPED
[INFO] spark-listeners-loganalytics_3.0.1_2.12 ............ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.699 s
[INFO] Finished at: 2021-03-11T17:05:33-08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M2:enforce (enforce-only-one-profile-is-active) on project spark-monitoring: org.apache.maven.plugins.enforcer.EvaluateBeanshell failed with message:
[ERROR] The expression "String activeProfiles = "[Profile {id: scala-2.12_spark-3.0.1, source: pom}, Profile {id: ossrh, source: settings.xml}]";
[ERROR]                                         int indexOf =activeProfiles.indexOf("Profile");
[ERROR]                                         if (indexOf == -1) {
[ERROR]                                             print("No active profile");
[ERROR]                                             return false;
[ERROR]                                         }
[ERROR]
[ERROR]                                         if (indexOf != activeProfiles.lastIndexOf("Profile")) {
[ERROR]                                             print("Only one profile may be active");
[ERROR]                                             return false;
[ERROR]                                         }
[ERROR]
[ERROR]                                         return true;" is not true.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**Comment by hallihan** [comment]
It looks like a second profile is active, being loaded from `settings.xml`, and this project is set to enforce only a single profile being active.

Can you check your ~/.m2/settings.xml and see if you have something like:
```xml
<settings>
  <profiles>
    <profile>
      <id>ossrh</id>
      <activation>
        <activeByDefault>true</activeByDefault>
      </activation>
```

and see if you can get the build to work by changing activeByDefault to false temporarily?

**Comment by brunoborges** [comment]
Alright, by adjusting profiles, it worked.

Is it really necessary to have such verification? I mean, how often will people try to build with multiple spark profiles activated? And wouldn't that fail?

**Comment by nfx** [comment]
i think that some CI system should just build it for different versions of spark and publish it.

**Comment by hallihan** [comment]
@brunoborges I'm tracking updates to the build on our internal backlog, including the use of maven-enforcer-plugin and the direct use of expressions in artifactId and version parameters.  I think the initial issues here were addressed in #130 - feel free to reopen with additional details if something further needs addressed here.
